lms/services/application_instance_getter.py

```diff
@@ -0,0 +1,117 @@
+from Crypto.Cipher import AES
+from sqlalchemy.orm.exc import NoResultFound
+
+from lms.services import ConsumerKeyError
+from lms.models import ApplicationInstance
+
+
+class ApplicationInstanceGetter:
+    """Methods for getting properties from application instances."""
+
+    def __init__(self, _context, request):
+        self._db = request.db
+        self._aes_secret = request.registry.settings["aes_secret"]
+
+    def developer_key(self, consumer_key):
+        """
+        Return the Canvas developer key for the given consumer_key, or None.
+
+        :arg consumer_key: the consumer key to search for
+        :type consumer_key: str
+
+        :raise ConsumerKeyError: if the consumer key isn't in the database
+
+        :return: the matching Canvas API developer key or ``None``
+        :rtype: str or ``None``
+        """
+        return self._get(consumer_key).developer_key
+
+    def developer_secret(self, consumer_key):
+        """
+        Return the Canvas developer secret for the given consumer_key, or None.
+
+        :arg consumer_key: the consumer key to search for
+        :type consumer_key: str
+
+        :raise ConsumerKeyError: if the consumer key isn't in the database
+
+        :return: the matching Canvas API developer secret or ``None``
+        :rtype: str or ``None``
+        """
+        application_instance = self._get(consumer_key)
+
+        developer_secret = application_instance.developer_secret
+        if developer_secret is None:
+            return None
+
+        cipher = AES.new(
+            self._aes_secret, AES.MODE_CFB, application_instance.aes_cipher_iv
+        )
+        return cipher.decrypt(developer_secret)
+
+    def lms_url(self, consumer_key):
+        """
+        Return the LMS URL for the given LTI consumer_key.
```

Comment:
No "or None" here because lms_url is always present. Some of the methods below are the same